src/Metadata/Resource/Factory/XmlResourceMetadataFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Annotation\Resource;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Metadata\Resource\ConfigurationMap;
+
+/**
+ * Creates a resource metadata from xml {@see Resource} configuration.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class XmlResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    use ResourceMetadataFactoryTrait;
+
+    private $configurationMap;
+    private $decorated;
+
+    public function __construct(ConfigurationMap $configurationMap, ResourceMetadataFactoryInterface $decorated = null)
+    {
+        $this->configurationMap = $configurationMap;
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass) : ResourceMetadata
+    {
+        $parentResourceMetadata = null;
+        if ($this->decorated) {
+            try {
+                $parentResourceMetadata = $this->decorated->create($resourceClass);
+            } catch (ResourceClassNotFoundException $resourceNotFoundException) {
+                // Ignore not found exception from decorated factories
+            }
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($resourceClass);
+        } catch (\ReflectionException $reflectionException) {
+            return $this->handleNotFound($parentResourceMetadata, $resourceClass);
+        }
+
+        if (!isset($metadata['shortName'])) {
+            $metadata['shortName'] = $reflectionClass->name;
+        }
+
+        $resource = new Resource();
```

Comment:
1. ofc
2. To avoid code repetition in `{Annotation,Xml,Yaml}ResourceMetadataFactory`? `AnnotationResourceMetadataFactory` is using an Object to build metadata. Btw I reverted the `ApiPlatform\Core\Annotation\Resource` file to the original value.